src/App.js

```diff
@@ -78,17 +80,45 @@ const App = () => {
 
   const resetGame = () => {
     // Complete in Wave 4
+    setSquares(generateSquares());
+    setCurrentPlayer(player1);
+    setWinningPlayer('...');
   };
 
+  const onClickCallback = (id) => {
+    if (winningPlayer !== '...') {
+      return;
+    }
+    setSquares((previous) => {
+      return previous.map((row) =>
+        row.map((square) => {
+          if (square.id === id && square.value === '') {
+            square.value = currentPlayer;
```

Comment:
Good call on mapping the two layers of lists. This makes a new outer list, and new row lists. However, there's still state cross contamination occurring as a result of editing square directly. We should make a copy (often by spreading) of the square. As it is, since both the old row and the new row refer to the same square data, we end up modify both the old state and the new state.